ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h

```diff
@@ -443,13 +444,19 @@ class HttpOperation
    * @param sockfd
    * @param for_recv
    * @param timeout_ms
-   * @return
+   * @return true if expected events occur, false if timeout or error happen
    */
-  static int WaitOnSocket(curl_socket_t sockfd, int for_recv, long timeout_ms)
+  static bool WaitOnSocket(curl_socket_t sockfd, int for_recv, long timeout_ms)
   {
+    bool res = false;
+
+#if defined(_WIN32)
+
+    if (sockfd > FD_SETSIZE)
```

Comment:
Should we retry several times when the `sockfd` is not selected? It usually happens when there is a lot actived fds, and `select` can not get them out once.

Reply:
Our service is quit busy.
Even though I set the trace sample_ration=0.01, the service startup several minutes latter, the CURLINFO_LASTSOCKET return sock fd above FD_SETSIZE. After this, all the trace message can not be upload and never recovered.
I don't think retry can work.